Source/Core/VideoBackends/Software/SWVertexLoader.cpp

```diff
@@ -32,32 +32,25 @@ SWVertexLoader::SWVertexLoader() = default;
 
 SWVertexLoader::~SWVertexLoader() = default;
 
+DataReader SWVertexLoader::PrepareForAdditionalData(OpcodeDecoder::Primitive primitive, u32 count,
+                                                    u32 stride, bool cullall)
+{
+  // Prevent multiple groups of vertices from being batched, as that causes issues with SetupUnit
+  Flush();
+  return VertexManagerBase::PrepareForAdditionalData(primitive, count, stride, cullall);
+}
+
```

Comment:
Unfortunately this causes problems, as (at least as far as I can tell) fifologs have the memory update after _all_ primitive data, as they get recorded when textures are loaded, and textures get loaded in the original `Flush()` call which is at the end of the batch.  Thus, the first object that uses a texture won't have the texture loaded, resulting in most of the fifoci differences seen here.  I don't think there's a way of avoiding this.

Reply:
I've removed the relevant commit.  (It's archived [here](https://github.com/pokechu22/dolphin/tree/archive/better-fifo-analyzer-part-3-sw-setupunit-raw-primitive-data)).  There are now no fifoci differences.